src/Serializer/AddressNormalizer.php

```diff
@@ -28,20 +31,20 @@ public function supportsDenormalization($data, $type, $format = null) : bool
     public function normalize($object, $format = null, array $context = []) : array
     {
         $data = [
-            'formatted' => $object->getFormatted(),
+            'formatted' => $object->getFormatted()->toArray(),
             'components' => [],
         ];
 
-        if ($object->getStreetAddress()) {
-            $data['components']['streetAddress'] = $object->getStreetAddress();
+        if (!$object->getStreetAddress()->isEmpty()) {
```

Comment:
If it were `if (!$streetAddress->isEmpty())` I'd say it's ok, but the `!` is a bit disconnected here. Maybe `notEmpty()`?